Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,261 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+inline std::ostream& operator<<(std::stringstream& file,const std::vector<Dir>& s)
+{
+    file << " " << s.size() << " ";
+    for (auto it : s)
+        file << " " << it << " ";
+    return file;
+}
+
+inline std::istream& operator>>(std::stringstream& file,std::vector<Dir>& s)
+{
+    size_t size;
+    file >> size;
+    s.reserve(size);
+
+    unsigned int local_id;
+    for (size_t i = 0; i < size; ++i)
+    {
+        file >> local_id;
+        s.push_back(local_id);
+    }
+    return file;
+}
+
+unsigned int hash(const std::vector<Dir>& content)
+{
+    unsigned int retval = 0;
+    int i = 1;
+    for (auto it = content.begin(); it != content.end(); ++it, ++i)
+    {
+        retval += *it * i;
+    }
+    return retval;
+}
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+    current_step_ = 0;
+    set_target_ = 0;
+    tickSpeed = 1;
+    pixSpeed = 4;
+}
```

Comment:
`reached_target` variable is not setted in constructor. Unsyncs dumps show that this variable cause the unsync.